lib/src/Pos/Launcher/Resource.hs

```diff
@@ -301,6 +309,7 @@ allocateNodeContext ancd txpSettings = do
             , ncNodeParams = np
             , ncTxpGlobalSettings = txpSettings
             , ncNetworkConfig = networkConfig
+            , ncMisbehaviorMetrics = Just mm
```

Comment:
I might be missing something, but it looks like `ncMisbehaviorMetrics` is always `Just`. Why is it `Maybe`?

Reply:
`Pos.Reporting.MemState` has `emptyReportingContext :: ReportingContext`, which is used in some tests and other things that I haven't dived into. That's where `Nothing` for misbehavior metrics is used.

Reply:
Ah, so it's needed for `setter`, ok!